File: misc/contains-duplicate-ii.c
bool containsNearbyDuplicate(int* nums, int numsSize, int k) {
    int i,j;
    for(i = 0; i < numsSize; ++i)
        for(j = i; j <= i+k && j < numsSize; ++j){
            if(i != j && nums[i] == nums[j])
                return true;
        }
        
    return false;
}

File: go/array/contains-duplicate-ii.go
func containsNearbyDuplicate(nums []int, k int) bool {
    for i := 0; i < len(nums); i++ {
        for j := i + 1; j < len(nums) && j <= i + k; j++{
            if(nums[i] == nums[j]){
                return true
            }
        }
    }

    return false
}
